### PR TITLE
Introduce ElementLocator to replace findElement overloads

### PR DESCRIPTION
**Pull request by tristanlabelle**
This brings a few improvements:
- Less code duplication
- Separation of accessibility id support in the WinAppDriver module
- Extensibility to other WebDriver backends
- Easier to implement higher-level functions like `expectElement` discussed elsewhere.

### CODE REVIEW

**Review by jeffdav — APPROVED** [review]
Looks great.

File: Sources/WebDriver/ElementLocator.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/// A locator strategy to use when searching for an element.
+public struct ElementLocator: Codable, Hashable {
+    /// The locator strategy to use.
+    public var using: String
```
[the remark this answers is not in the document]
Comment (tristanlabelle) [reply Sources/WebDriver/ElementLocator.swift:4]: I started that way but it doesn't allow adding cases for accessibilityId from the WinAppDriver module, or extending to other WebDriver backend-specific strategies.

File: Sources/WebDriver/ElementLocator.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    /// The search target.
+    public var value: String
+
+    public init(using: String, value: String) {
```
Comment (vinocher-bc) [inline Sources/WebDriver/ElementLocator.swift:8]: For my understanding, could you please explain why the accessibilityId factory method was not added here, and was instead added in WinAppDriver\ElementLocator+accessibilityId.swift? Adding it here could have made the initializer private, forcing creation only by using factory methods.

Comment (tristanlabelle) [reply Sources/WebDriver/ElementLocator.swift:8]: The "accessibility id" strategy is not part of the WebDriver spec but an extension of WinAppDriver. We want to allow usage of this library with arbitrary webdriver backends that might support other private strategies, so we need the public initializer.

**Pull request merged**